api/apps/restful_apis/file2document_api.py

```
@@ -33,26 +33,28 @@
-                logging.warning("tenant_id not found for doc_id=%s, skipping remove_document", doc_id)
-                continue
-            DocumentService.remove_document(doc, tenant_id)
-        File2DocumentService.delete_by_file_id(id)
+    """Synchronous worker: add new docs for the given file/kb pairs while preserving existing links.
 
+    Previously this function replaced all existing links with the new ones, which caused
+    multi-select "link to knowledge base" to overwrite previous links. Now it only creates
+    documents for knowledge bases that are not already linked to the file, and leaves
+    existing links untouched.
+    """
+    for id in file_ids:
```

Ruff (0.15.20)

[error] Variable `id` is shadowing a Python builtin

(A001)